### ISSUE TITLE
feat: Implement focus trap for nav, content, and file elements

### ISSUE THREAD

**Issue by katie-z-geer** [zed-industries/zed#42673]
### Summary

**Summary**

Implement a manual focus trap for the navigation, content, and file elements in the UI. Tabbing at the end of one section should wrap focus around to the beginning of that same section, rather than moving to the next UI element.

**Description**

To prevent focus from being lost or moving to unexpected parts of the UI, a focus trap should be implemented manually for the main layout components. This will ensure that when a user tabs through all the elements in the navigation pane, for example, the focus will loop back to the first element in the navigation pane.

**Note:**
Implementing this behavior is expected to fix the bug where tabbing off-screen causes focus to be lost.

### Zed Version and System Specs

n/a

### If applicable, attach your `Zed.log` file to this issue.

<details><summary>Zed.log</summary>

```log

```

</details>
